core/src/test/java/uk/gov/gchq/koryphe/impl/predicate/range/AbstractInTimeRangeDualTest.java

```diff
@@ -156,6 +168,94 @@ public void shouldRejectValuesNotInExclusiveRange() throws IOException {
         testValues(false, values, filter);
     }
 
+    @Test
+    public void shouldAcceptValuesInStartAndEndPartiallyContained() throws IOException {
+        // Given
+        final Predicate filter = createBuilder()
+                .start("1")
+                .end("10")
+                .startInclusive(false)
+                .endInclusive(false)
+                .startFullyContained(false)
+                .endFullyContained(false)
+                .build();
+
+        final List<Tuple2<Long, Long>> values = Arrays.asList(
+                new Tuple2<>(1L, 10L),
+                new Tuple2<>(1L, 5L),
+                new Tuple2<>(5L, 10L)
+        );
+
+        // When / Then
+        testValues(true, values, filter);
+    }
+
+    @Test
+    public void shouldAcceptValuesInStartAndEndPartiallyContainedInclusive() throws IOException {
+        // Given
+        final Predicate filter = createBuilder()
+                .start("1")
+                .end("10")
+                .startInclusive(true)
+                .endInclusive(true)
+                .startFullyContained(false)
+                .endFullyContained(false)
+                .build();
+
+        final List<Tuple2<Long, Long>> values = Arrays.asList(
+                new Tuple2<>(0L, 10L),
+                new Tuple2<>(0L, 5L),
+                new Tuple2<>(5L, 11L),
+                new Tuple2<>(10L, 10L)
+        );
+
+        // When / Then
+        testValues(true, values, filter);
+    }
+
+    @Test
+    public void shouldRejectValuesInStartAndEndPartiallyContained() throws IOException {
+        // Given
+        final Predicate filter = createBuilder()
+                .start("1")
+                .end("10")
+                .startInclusive(false)
+                .endInclusive(false)
+                .startFullyContained(false)
+                .endFullyContained(false)
+                .build();
+
+        final List<Tuple2<Long, Long>> values = Arrays.asList(
+                new Tuple2<>(0L, 1L),
+                new Tuple2<>(10L, 11L)
+        );
+
+        // When / Then
+        testValues(false, values, filter);
+    }
+
+    @Test
+    public void shouldRejectValuesInStartAndEndPartiallyContainedInclusive() throws IOException {
+        // Given
+        final Predicate filter = createBuilder()
+                .start("1")
+                .end("10")
+                .startInclusive(true)
+                .endInclusive(true)
+                .startFullyContained(false)
+                .endFullyContained(false)
+                .build();
+
+        final List<Tuple2<Long, Long>> values = Arrays.asList(
+                new Tuple2<>(-1L, 0L),
+                new Tuple2<>(11L, 12L)
+        );
+
+        // When / Then
+        testValues(false, values, filter);
+    }
+
+
     @Test
     public void shouldJsonSerialiseAndDeserialisWithExclusive() throws IOException {
```

Comment:
Maybe include the startFullyContained and endFullyContained within the shouldJsonSerialiseAndDeserialise method?